common/admin/smb/enums.go

```diff
@@ -0,0 +1,117 @@
+//go:build !(octopus || pacific || quincy || reef || squid) && ceph_preview
+
+package smb
+
+// Intent indicates how a resource description should be processed.
+type Intent string
+
+const (
+	// Present resources will be created or updated.
+	Present = Intent("present")
```

Comment:
Probably has no real-world effect, but want to mention it anyway: like `errors.New()` using a pointer to a string instead of the string itself makes the switch-cases faster and allows identical string in different enums.

Reply:
I don't think I would want to change it now, but I will keep that in mind for the future.